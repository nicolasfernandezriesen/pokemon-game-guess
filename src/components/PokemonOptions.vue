<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokemons" :key="pokemon.id"
      @click="$emit('selectedPokemon', pokemon.id)">
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
  ul {
      list-style-type: none;
  }
  li {
      background-color: rgb(93, 50, 248);
      border-radius: 5px;
      border: 1px solid rgba(0, 0, 0, 0.2);
      cursor: pointer;
      margin-bottom: 10px;
      width: 250px;
      color: black;
  }

  li:hover {
      background-color: rgba(89, 46, 246, 0.652);
  }

  .options-container {
      display: flex;
      justify-content: center;
  }
</style>