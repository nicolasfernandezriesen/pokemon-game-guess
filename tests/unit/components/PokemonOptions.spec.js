import { shallowMount } from "@vue/test-utils";
import PokemonOptions from '@/components/PokemonOptions'
import {pokemons} from '../mocks/pokemons.mock'; 

describe ('PokemonOptions', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons
            }
        });
    });
    test('it should match with the snapshot', () => {
        //Arrange
        //Act
        //Assert
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('it should show the 4 options', () => {
        let lis = wrapper.findAll('li');

        expect(lis.length).toBe(4);

        expect(lis[0].text()).toBe('bulbasaur');
        expect(lis[1].text()).toBe('ivysaur');
        expect(lis[2].text()).toBe('venusaur');
        expect(lis[3].text()).toBe('charmander');
    })

    test('it should emit the event with the correct parameter', () => {
        //Arrange
        let [li1, li2, li3, li4] = wrapper.findAll('li');
        
        //Act
        li1.trigger('click');

        //Assert
        expect(wrapper.emitted('selectedPokemon').length).toBe(1);
        expect(wrapper.emitted('selectedPokemon')[0]).toEqual([1]);
    })
})