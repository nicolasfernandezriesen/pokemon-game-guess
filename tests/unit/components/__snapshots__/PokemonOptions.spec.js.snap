// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions it should match with the snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
